src/workbookManager.ts

```diff
@@ -40,6 +40,28 @@ export const generateSingleQueryWorkbook = async (query: QueryInfo, initialDataG
     return await generateSingleQueryWorkbookFromZip(zip, query, fileConfigs, tableData);
 };
 
+export const generateMultipleQueryWorkbook = async (queries: MultipleQueriesInfo, initialDataGrid?: Grid, fileConfigs?: FileConfigs): Promise<Blob> => {
+    const templateFile: File | undefined = fileConfigs?.templateFile;
+    if (templateFile !== undefined && initialDataGrid !== undefined) {
+        throw new Error(templateWithInitialDataErr);
+    }
+
+    if (!queries.loadedQuery.queryName) {
+        queries.loadedQuery.queryName = defaults.queryName;
+    }
+
+    const zip: JSZip =
+        templateFile === undefined ? await JSZip.loadAsync(SIMPLE_QUERY_WORKBOOK_TEMPLATE, { base64: true }) : await JSZip.loadAsync(templateFile);
+
+    const tableData = initialDataGrid ? gridUtils.parseToTableData(initialDataGrid) : undefined;
+
+    const connectionOnlyQueryNames: string[] = pqUtils.validateMultipleQueryNames(queries.connectionOnlyQueries, queries.loadedQuery.queryName!);
+    pqUtils.validateQueryName(queries.loadedQuery.queryName!);
```

Comment:
better to move the argument validation to the top of the function (early return)